pkgs/development/tools/misc/blackmagic/default.nix

```diff
@@ -1,36 +1,38 @@
 { stdenv, lib, fetchFromGitHub
-, gcc-arm-embedded, binutils-arm-embedded, libftdi
+, gcc-arm-embedded, libftdi1
 , python, pythonPackages
 }:
 
 with lib;
 
 stdenv.mkDerivation rec {
   name = "blackmagic-${version}";
-  version = "1.6.1";
+  # use `git describe --always | tail -c +2` to match upstream version
+  # format in flashed firmware
+  version = "1.6.1-311-gfbf1963";
```

Comment:
Is it important to match the version format in flashed firmware?
Otherwise, the convention to use for unstable software versions (commits from master for example) is
```
version = "unstable-YYYY-MM-DD";
```

in nixpkgs. The date being that of the commit.

Reply:
Oh I'm seeing it used at line 35. Can you add an attribute like `firmwareVersion` with this and have `version` be what I've suggested?